internal/database/sqlcommon/chart_sql.go

```diff
@@ -86,174 +47,116 @@ func (s *SQLCommon) getTableNameFromCollection(ctx context.Context, collection d
 	}
 }
 
-func (s *SQLCommon) getDistinctTypesFromTable(ctx context.Context, tableName string, fieldMap map[string]string) ([]string, error) {
-	if _, ok := fieldMap["type"]; !ok {
-		return []string{}, nil
-	}
-	qb := sq.Select(fieldMap["type"]).Distinct().From(tableName)
-
-	rows, _, err := s.query(ctx, qb.From(tableName))
-	if err != nil {
-		return nil, err
-	}
-	defer rows.Close()
-
-	var dataTypes []string
-	for rows.Next() {
-		var dataType string
-		err := rows.Scan(&dataType)
-		if err != nil {
-			return []string{}, i18n.WrapError(ctx, err, i18n.MsgDBReadErr, tableName)
-		}
-		dataTypes = append(dataTypes, dataType)
+func (s *SQLCommon) getSelectStatements(ns string, tableName string, intervals []fftypes.ChartHistogramInterval, timestampKey string, sql sq.SelectBuilder) (queries []sq.SelectBuilder) {
+	for _, interval := range intervals {
+		queries = append(queries, sql.
+			From(tableName).
+			Where(
+				sq.And{
+					sq.GtOrEq{timestampKey: interval.StartTime},
+					sq.Lt{timestampKey: interval.EndTime},
+					sq.Eq{"namespace": ns},
+				}).
+			OrderBy(timestampKey).
+			Limit(uint64(config.GetInt(config.DatabaseMaxChartRows))))
 	}
-	rows.Close()
 
-	return dataTypes, nil
+	return queries
 }
 
-func (s *SQLCommon) getBucketTotal(typeBuckets []*fftypes.ChartHistogramType) (string, error) {
+func (s *SQLCommon) histogramResult(ctx context.Context, tableName string, rows *sql.Rows, onlyTimestamp bool) (map[string]int, string, error) {
```

Comment:
Is there a reason this returns `map[string]int, string, error` and not `map[string]int, int, error`? It seems like it's always empty or a number

Reply:
The API expects strings and not integers in the response, so this function returns a string for the `total` value so that it doesn't have to be converted in `GetChartHistogram()`